status-page.py

```diff
@@ -7,63 +8,68 @@
 
 class StatusPage:
     def __init__(self):
-        config_data_file = open(sys.argv[1])
-        log_file = open(sys.argv[2])
-        self.config_json = json.load(config_data_file)
-        self.log_json = json.load(log_file)
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            '--config',
+            dest='config_path',
+            help='Path to configuration.json',
+            required=True
+        )
+        parser.add_argument(
+            '--status',
+            dest='api_status_path',
+            help='Path to api_status.json',
+            required=True
+        )
+        args, unknown_args = parser.parse_known_args()
+        self.config_json = json.load(open(args.config_path))
+        self.api_status_json = json.load(open(args.api_status_path))
         self.access_token = self.config_json['access_token']
         self.base_url = self.config_json['base_url']
+        self.api_map = self.config_json['api_map']
         self.header = {'x-cachet-token': self.access_token}
         self.broken_components = []
         self.open_incidents = []
 
     def get_broken_components(self):
-        response = requests.get(f'{self.base_url}/components', self.header)
+        """Fetches all apis not marked as 'Operational'"""
+        try:
+            response = requests.get(f'{self.base_url}/components', self.header)
+        except requests.exceptions.RequestException as err:
+            print(f'REQUEST ERROR! Unable to get components.\n{err}')
+            sys.exit(1)
         for component in response.json()['data']:
-            status = component['status']
-            if status != 1:
+            if component['status'] != 1:
                 self.broken_components.append(component)
 
     def get_all_open_incidents(self):
-        response = requests.get(f'{self.base_url}/incidents', self.header)
+        """Fetches all incidents not marked as 'Resolved'"""
+        try:
+            response = requests.get(f'{self.base_url}/incidents', self.header)
```

Comment:
I guess you can use [`requests.session`](https://requests.readthedocs.io/en/master/user/advanced/#session-objects) here so that you don't need to pass header for all requests.

Example: https://github.com/osu-mist/express-api-skeleton/blob/master/src/tests/integration/utils.py#L66